xarray/backends/zarr.py

```diff
@@ -623,7 +623,12 @@ def store(
             # avoid needing to load index variables into memory.
             # TODO: consider making loading indexes lazy again?
             existing_vars, _, _ = conventions.decode_cf_variables(
-                self.get_variables(), self.get_attrs()
+                {
```

Comment:
feels like we should also be skipping this for `mode="w"` to allow overwriting the existing encoding?

Reply:
Can you expand on what you mean? Is that because we would have just written these vars so already know their values/schema?

Reply:
`mode="w"` means overwrite so we shoudn't care about encoding on disk, no?

Reply:
I added a test, the encoding does get update as expected with mode="w", so presumably zarr is nuking the store with mode="w".